src/main/java/org/jenkinsci/plugins/github_branch_source/Endpoint.java

```diff
@@ -115,14 +116,27 @@ public FormValidation doCheckApiUri(@QueryParameter String apiUri) {
                 URL api = new URL(apiUri);
                 GitHub github = GitHub.connectToEnterpriseAnonymously(api.toString());
                 github.checkApiUrlValidity();
-                return FormValidation.ok();
+                LOGGER.log(Level.FINE, "Trying to configure a GitHub Enterprise server");
+                return FormValidation.ok("GitHub Enterprise server verified");
             } catch (MalformedURLException mue) {
-                return FormValidation.error("This does not look like a GitHub Enterprise API URL");
+                // For example: https:/api.github.com
+                LOGGER.log(Level.WARNING, "Trying to configure a GitHub Enterprise server: " + apiUri);
+                return FormValidation.error("This does not look like a GitHub Enterprise API endpoint");
             } catch (JsonParseException jpe) {
-                return FormValidation.error("This does not look like a GitHub Enterprise API URL");
+                LOGGER.log(Level.WARNING, "Trying to configure a GitHub Enterprise server: " + apiUri);
+                return FormValidation.error("This does not look like a GitHub Enterprise API endpoint");
+            } catch (FileNotFoundException fnt) {
+                // For example: https://github.mycompany.com/server/api/v3/ gets a FileNotFoundException
+                LOGGER.log(Level.WARNING, "Getting HTTP Error 404 for " + apiUri);
+                return FormValidation.error("This does not look like a GitHub Enterprise API endpoint");
```

Comment:
Yes, I understood that;  each one of these error cases should give a different formvalidation error and a different log message, so that the user/Jenkins administrator knows which thing is the problem.   For the 404 case, you just need to say: "This does not look like a GitHub Enterprise API endpoint: page not found (error 404)" + apiUrl